### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by Rachelwebdev**
### In this milestone, I:

- Fixed N+1 problem when fetching all posts and their comments for a user.

**before the fix:** 
N + 1 rails 1

**After the fix**

N + 1 rails 2

- Used Capybara to write integration tests for:

1. User index page
2. user show page
3. User post index page
4. Post show page

- Fixed linter errors

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
     @posts = Post.all
+    @user = User.includes(:posts, :comments).find(params[:user_id])
```
Comment (bobb-Rob) [inline app/controllers/posts_controller.rb]: - [ ] This `index` action should use the `.includes` methods on the `@posts` like so 👇🏾 

```ruby
class PostsController < ApplicationController
  def index
    @user = User.find(params[:user_id])
    @posts = @user.posts.includes(:comments)   
  end
# --- other actions
end
```

Kindly refactor your code as above.

File: spec/system/users/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  it 'displays number of posts each user has written' do
+    expect(page).to have_content "Number of posts: #{@user2.post_count}"
+  end
+
+  it 'redirects to to a users show page when a user is clicked' do
+    click_on @user2.name
+    expect(page).to have_current_path user_path(@user2.id)
+  end
+end
```
Comment (bobb-Rob) [inline spec/system/users/index_spec.rb]: - [ ] On this users_index_spec test, the test case -`I can see the username of all other users.` is missing the assertion of the second user's username. Similarly, the test case `I can see the profile picture for each user` is missing assertion for the second user's image. _Please use the code snippet below as a guide to improve your test suites, please `❗ note` that this is not an actual implementation._ 

```ruby
require 'rails_helper'

RSpec.feature "User index page", type: :feature do
  before(:each) do
    @user1 = User.create!(username: "user1", password: "password")
    @user2 = User.create!(username: "user2", password: "password")

    @post1 = Post.create!(title: "Post 1", body: "Body 1", user_id: @user1.id)
    @post2 = Post.create!(title: "Post 2", body: "Body 2", user_id: @user1.id)
    @post3 = Post.create!(title: "Post 3", body: "Body 3", user_id: @user2.id)
  end

  scenario "User can see the username of all other users" do
    visit users_path

    expect(page).to have_content(@user1.username)
    expect(page).to have_content(@user2.username)
  end

  scenario "User can see the profile picture for each user" do
    visit users_path

    expect(page).to have_css("img[src*='#{@user1.profile_picture_url}']")
    expect(page).to have_css("img[src*='#{@user2.profile_picture_url}']")
  end

  scenario "User can see the number of posts each user has written" do
    visit users_path

    expect(page).to have_content("Posts written by #{@user1.username}: 2")
    expect(page).to have_content("Posts written by #{@user2.username}: 1")
  end

  scenario "When I click on a user, I am redirected to that user's show page" do
    visit users_path

    click_link(@user1.username)

    expect(current_path).to eq(user_path(@user1))
  end
end
```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Rachelwebdev, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @Rachelwebdev 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**